### ISSUE TITLE
Joi.func() creates Objects as examples

### ISSUE THREAD

**Issue by MrShoenel** [xogroup/felicity#141]
#### Context

* *node version*: latest
* *felicity version*: 2.4.7
* *environment* (node, browser): node
* *any relevant modules used with* (hapi, hoek, etc.):
* *any other relevant information*:

#### What are you trying to achieve or the steps to reproduce ?

When trying to generate valid examples that include schemas of type `Joi.func()`, felicity creates objects.

#### What result did you expect ?

To have a function instead of an object.

#### What result did you observe ?

I got a plain object.

#### Reproduce:

```
const joiSchema = Joi.object().keys({
  foo: Joi.func()
});

joiSchema.validate({ foo: function(){} }); // error === null
const FelicityConstructor = Felicity.entityFor(joiSchema);
const felicityInstance = new FelicityConstructor();
let felicityExample = felicityInstance.example();

console.log(typeof felicityExample.foo); // "object"
```

**Comment by WesTyler** [comment]
Ah weird! Thanks for the report, I confirmed it is reproducable :)

I'll see if I can get a patch out this weekend.

**Comment by WesTyler** [comment]
Huh. Wonder why github didn't close this out with that commit. 

This should be fixed in v2.4.8! Thanks again for checking out Felicity and taking the time to report the problem!

Please let me know if you run into anything else! :D
